kobo/apps/subsequences/utils/deprecation.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import annotations
+
+from copy import deepcopy
+from typing import Optional
+
+import jsonschema
+
+from kpi.fields import WritableJSONField
+from ..advanced_features_params_schema import (
+    ADVANCED_FEATURES_PARAMS_SCHEMA,
+)
+
+
+def jsonschema_validate(asset: 'Asset'):
+    try:
+        jsonschema.validate(
+            instance=asset.advanced_features,
+            schema=ADVANCED_FEATURES_PARAMS_SCHEMA,
+        )
+    except jsonschema.exceptions.ValidationError as e:
+        if "'qpath' was unexpected" not in str(e):
+            raise
+
+        qual_survey_orig = asset.advanced_features['qual']['qual_survey']
+        qual_survey_iter = deepcopy(qual_survey_orig)
+        for idx, qual_q in enumerate(qual_survey_iter):
+            qpath = qual_survey_orig[idx]['qpath']
```

Comment:
Should check if 'qpath' key exists and continue if not. If it was sanitized before, qpath won't be present, right?

Reply:
Indeed, `jsonschema.validate()` does raise an error if `qpath` is present because it does not belong to `ADVANCED_FEATURES_PARAMS_SCHEMA`. 

Line 24 cannot be reached if `qpath` is not present.